pkg/auth/jwt.go

```diff
@@ -0,0 +1,68 @@
+package auth
+
+import (
+	"fmt"
+	"net/http"
+
+	"NKSS-backend/pkg/handlers"
+
+	"github.com/golang-jwt/jwt"
+	_ "github.com/lib/pq"
+)
+
+func Authenticator(next http.Handler, secret string) http.Handler {
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		// Check authorization via JWT
+		tokenStr := r.Header.Get("Authorization")
+		if tokenStr == "" || ValidateJWT(tokenStr, secret) == "null" {
+			handlers.RespondJSON(w, 401, "Authorization Error")
+			return
+		}
+		next.ServeHTTP(w, r)
+	})
+}
+
+func CreateJWT(role string, rollno string, secret string) string {
+
+	hmacSecret := []byte(secret)
+
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"role":   role,
+		"rollno": rollno,
+	})
+
+	tokenString, err := token.SignedString(hmacSecret)
+
+	if err != nil {
+		fmt.Println("An unexpected error occurred: ", err)
+		return ""
+	} else {
+		return tokenString
+	}
+
+}
+
+func ValidateJWT(tokenString string, secret string) string {
+
+	hmacSecret := []byte(secret)
+
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		}
+
+		return hmacSecret, nil
+	})
+	fmt.Println(err)
+
+	claims, ok := token.Claims.(jwt.MapClaims)
+
+	if ok && token.Valid {
+		return string(claims["rollno"].(string))
+	} else {
+		return "null"
```

Comment:
NO! should return a `(string, boolean)`

Reply:
Okay, have made the function a bit more verbose like this